src/services/bundle_repository.js

```diff
@@ -78,14 +80,33 @@ export default class BundleRepository {
     return await downloadJSONFromGridFSBucket(bundleId, this.bundlesBucket);
   }
 
-  async createBundleMetadata(bundleId, storagePeriods, status = BundleStatusStates.unknown) {
+  async createBundleMetadata(bundleId, storagePeriods, status = BundleStatuses.unknown, additionalRepositoryFields = {}) {
     if (await this.db.collection('bundle_metadata').findOne({bundleId}) === null) {
       await this.db.collection('bundle_metadata').insertOne({
-        bundleId, storagePeriods, repository: {status}
+        bundleId, storagePeriods, repository: {...additionalRepositoryFields, status}
       });
     }
   }
 
+  async storeAdditionalMetadata(bundleId, complementaryMetadata) {
+    const requiredFields = [
+      'bundleId',
+      'bundleTransactionHash',
+      'bundleProofBlock',
+      'bundleUploadTimestamp',
+      'storagePeriods',
+      'repository'
+    ];
+    const additionalMetadataFields = pipe(...requiredFields.map((fieldName) =>
```

Comment:
I'm not sure if this is proper and needed `pipe` usage

Reply:
It does pick(pick(pick(pick(pick(data))))), exactly wat pipe does

Reply:
If pick was written as fieldname => dict => ... it would look much nicer